libbrpc.so: undefined reference to `_dl_sym'

**Describe the bug (描述bug)**

```
[ 94%] Linking CXX executable ../output/bin/protoc-gen-mcpack                                                                                                                        
[ 94%] Linking CXX executable ../../output/bin/parallel_http                                                                                                                         
/usr/bin/ld: ../output/lib/libbrpc.so: undefined reference to `_dl_sym'                                                                                                              
collect2: error: ld returned 1 exit status
make[2]: *** [src/CMakeFiles/protoc-gen-mcpack.dir/build.make:106: output/bin/protoc-gen-mcpack] Error 1
make[1]: *** [CMakeFiles/Makefile2:347: src/CMakeFiles/protoc-gen-mcpack.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
/usr/bin/ld: ../../output/lib/libbrpc.a(mutex.cpp.o): in function `bthread::init_sys_mutex_lock()':
/data/incubator-brpc/src/bthread/mutex.cpp:411: undefined reference to `_dl_sym'
/usr/bin/ld: /data/incubator-brpc/src/bthread/mutex.cpp:412: undefined reference to `_dl_sym'
collect2: error: ld returned 1 exit status
make[2]: *** [tools/parallel_http/CMakeFiles/parallel_http.dir/build.make:108: output/bin/parallel_http] Error 1
make[1]: *** [CMakeFiles/Makefile2:373: tools/parallel_http/CMakeFiles/parallel_http.dir/all] Error 2
```

**To Reproduce (复现方法)**


**Expected behavior (期望行为)**


**Versions (各种版本)**
OS: fedora 36
Compiler: gcc 12
brpc:
protobuf:

**Additional context/screenshots (更多上下文/截图)**


`_dl_sym` 不是在 libc.so 中吗，为啥没自动link呢

--- a/src/bthread/mutex.cpp
+++ b/src/bthread/mutex.cpp
@@ -44,7 +44,7 @@
 #include "bthread/log.h"
 
 extern "C" {
-extern void* _dl_sym(void* handle, const char* symbol, void* caller);
+extern void* __attribute__((weak)) _dl_sym(void* handle, const char* symbol, void* caller);
 }
 
 namespace bthread {
@@ -408,8 +408,14 @@ static void init_sys_mutex_lock() {
 #if defined(OS_LINUX)
     // TODO: may need dlvsym when GLIBC has multiple versions of a same symbol.
     // http://blog.fesnel.com/blog/2009/08/25/preloading-with-multiple-symbol-versions
-    sys_pthread_mutex_lock = (MutexOp)_dl_sym(RTLD_NEXT, "pthread_mutex_lock", (void*)init_sys_mutex_lock);
-    sys_pthread_mutex_unlock = (MutexOp)_dl_sym(RTLD_NEXT, "pthread_mutex_unlock", (void*)init_sys_mutex_lock);
+    if (_dl_sym) {
+        sys_pthread_mutex_lock = (MutexOp)_dl_sym(RTLD_NEXT, "pthread_mutex_lock", (void*)init_sys_mutex_lock);
+        sys_pthread_mutex_unlock = (MutexOp)_dl_sym(RTLD_NEXT, "pthread_mutex_unlock", (void*)init_sys_mutex_lock);
+    } else {
+        // _dl_sym may be undefined reference in some system, fallback to dlsym
+        sys_pthread_mutex_lock = (MutexOp)dlsym(RTLD_NEXT, "pthread_mutex_lock");
+        sys_pthread_mutex_unlock = (MutexOp)dlsym(RTLD_NEXT, "pthread_mutex_unlock");
+    }
 #elif defined(OS_MACOSX)
     // TODO: look workaround for dlsym on mac
     sys_pthread_mutex_lock = (MutexOp)dlsym(RTLD_NEXT, "pthread_mutex_lock");

